server/server.go

```diff
@@ -351,6 +344,26 @@ func (s *Server) Kill(connectionID uint64, query bool) {
 	}
 }
 
+// GracefulDown waits all clients to close.
+func (s *Server) GracefulDown() {
+	log.Info("graceful shut down.")
+	s.rwlock.RLock()
+	for _, conn := range s.clients {
+		conn.killed = true
+		atomic.StoreInt64(&conn.keepAlive, 42)
+	}
+	s.rwlock.RUnlock()
+
+	for s.ConnectionCount() > 0 {
+		time.Sleep(time.Second)
```

Comment:
Is one second long enough?

Reply:
Any suggestion to distinguish that the connection is block on `readPacket` rather than `dispatch`?
If `dispatch` runs more than a second, this method can't work.